util/whitelist/debugger_whitelist.go

```diff
@@ -0,0 +1,83 @@
+package whitelist
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/patrickmn/go-cache"
+	"github.com/sirupsen/logrus"
+)
+
+var (
+	whiteListURL   string
+	whiteListCache = cache.New(5*time.Minute, 5*time.Minute)
+)
+
+func init() {
+	whiteListURL = os.Getenv("WHITELIST_BACKEND_URL")
+	logrus.Info("whiteListURL: ", whiteListURL)
+}
+
+// IsIPValid checks if the debugger IP is in the whitelist through the whitelist backend
+func IsIPValid(ip string) (bool, error) {
+	if ip == "unknown_ip" {
+		return false, nil
+	}
+
+	cacheKey := "whitelist-ip-" + ip
+	cacheValue, found := whiteListCache.Get(cacheKey)
+	logrus.Debug("whitelist IP cache Get ip: ", ip, ", found: ", found, ", cacheValue: ", cacheValue)
+	if found {
+		return cacheValue.(bool), nil
+	}
+
+	params := url.Values{}
+	Url, err := url.Parse(whiteListURL + "api/get_debugger")
+	if err != nil {
+		return false, err
+	}
+	params.Set("accept", "application/json")
+	params.Set("ip", ip)
+	Url.RawQuery = params.Encode()
+	urlPath := Url.String()
+	resp, err := http.Get(urlPath)
+	if err != nil {
+		return false, err
+	}
+	defer resp.Body.Close()
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
```

Comment:
Here too, this normally should not happen.

Reply:
correct, fixed.